environment.yml

```diff
@@ -76,7 +76,7 @@ dependencies:
 
   # code checks
   - flake8=6.0.0  # run in subprocess over docstring examples
-  - mypy=1.2  # pre-commit uses locally installed mypy
+  - mypy=1.4.1  # pre-commit uses locally installed mypy
```

Comment:
Could you remind me why we use the locally install mypy instead of using the pre-commit mirror?

Reply:
I would like that, but I believe the issue was/is that we would need to re-install all the external libraries that are used inside pandas so that mypy can check whether pandas uses them correctly (possible but slower). I'm unsure whether that would be possible for pyright: pyright is a node application, pre-commit doesn't(?) allow a node-hook to have python dependencies.

Reply:
Ah right right. Yeah that makes sense, and I think is still the case.